src/Smalot/PdfParser/PDFObject.php

```diff
@@ -274,14 +275,43 @@ public function getText(Page $page = null)
                         $args = preg_split('/\s/s', $command[self::COMMAND]);
                         $y = array_pop($args);
                         $x = array_pop($args);
+                        // get max char size and use as minimum for horizontal offset
+                        $fontMaxWidth = 0;
+                        if (null !== $current_font) {
+                            $fontDictonary = $current_font->getDetails();
+                            // type 0
+                            if (isset($fontDictonary['DescendantFonts'])) {
+                                $fontDictonary = $fontDictonary['DescendantFonts'][0];
+                            }
+                            // type 1
+                            if (isset($fontDictonary['Widths'])) {
+                                foreach ($fontDictonary['Widths'] as $width) {
+                                    if ((float) $width > $fontMaxWidth) {
+                                        $fontMaxWidth = (float) $width;
+                                    }
+                                }
+
+                                $fontMaxWidth = ($fontMaxWidth / 1000) * $current_font_size;
+                            }
+                            // CIDFontType2
+                            if ('cidfonttype2' === strtolower($fontDictonary['Type'])) {
+                                if (isset($fontDictonary['DW']) && $fontDictonary['DW']) {
+                                    $fontMaxWidth = ((float) $fontDictonary['DW'] / 1000) * $current_font_size;
+                                } else {
+                                    // default
+                                    $fontMaxWidth = $current_font_size;
+                                }
+                            }
+                        }
+
                         if (((float) $x <= 0) ||
                             (false !== $current_position_td['y'] && (float) $y < (float) ($current_position_td['y']))
                         ) {
                             // vertical offset
                             $text .= "\n";
-                        } elseif (false !== $current_position_td['x'] && (float) $x > (float) (
+                        } elseif (false !== $current_position_td['x'] && ((float) $x - (float) (
                                 $current_position_td['x']
-                            )
+                            )) > $fontMaxWidth
```

Comment:
That is complicated to read. Can you move some computation into a variable and use it in the `elseif` instead?